dev-tools/globals.ts

```diff
@@ -219,7 +220,7 @@ interface EventMethods {
 	onEat?: ((this: Battle, pokemon: Pokemon) => void) | false
 	onEatItem?: (this: Battle, item: Item, pokemon: Pokemon) => void
 	onEffectiveness?: (this: Battle, typeMod: number, target: Pokemon | null, type: string, move: ActiveMove) => void
-	onEnd?: (this: Battle, pokemon: Pokemon & Side) => void
+	onEnd?: (this: Battle, target: Pokemon & Side & Field) => void
```

Comment:
Using `&` here is really sketchy - this should really be `|` because handlers are really expecting one or the other, not a combination of both.

Reply:
I assume the `&` is explicitly to shut up a TypeScript complaint that's really hard to deal with normally.

Reply:
I don't think so, unless by "TypeScript complaint" you mean the compiler is telling us "you're not using types properly at all". Try changing these to `|` and run `npm run tsc` - handlers just want to not bother checking what type of object theyve been given and want to go about making assumptions about what fields they have.

So yeah, `&` is definitely for convenience, but makes a farce of the typesystem because its not true.